test/Tester/PSClientTests/PSClientTests.cs

```diff
@@ -0,0 +1,145 @@
+            Runspace = RunspaceFactory.CreateRunspace(initialSessionState);
+            Runspace.Open();
+            Powershell = PowerShell.Create();
+            Powershell.Runspace = Runspace;
+
+            var stopGrainClient = new Command("Stop-GrainClient");
+            Powershell.Commands.AddCommand(stopGrainClient);
+            Powershell.Invoke();
+        }
+
+        public override void Dispose()
+        {
+            var stopCommand = new Command("Stop-GrainClient");
+            Powershell.Commands.Clear();
+            Powershell.Commands.AddCommand(stopCommand);
+            Powershell.Invoke();
+            Powershell.Dispose();
+            Runspace.Dispose();
+            base.Dispose();
+        }
+    }
+
+    public class PSClientTests : OrleansTestingBase, IClassFixture<PowershellHostFixture>
+    {
+        private PowerShell _ps;
+        private ClientConfiguration _clientConfig;
+        private ITestOutputHelper output;
+
+        public PSClientTests(ITestOutputHelper output, PowershellHostFixture fixture)
+        {
+            this.output = output;
+            _clientConfig = fixture.HostedCluster.ClientConfiguration;
+            _ps = fixture.Powershell;
+            _ps.Commands.Clear();
+        }
+
+        [Fact, TestCategory("BVT"), TestCategory("Tooling")]
+        public void ScriptCallTest()
+        {
+            _ps.Commands.AddScript(File.ReadAllText(@".\PSClientTests\PSClientTests.ps1"));
+            _ps.Commands.AddParameter("clientConfig", _clientConfig);
+            var results = _ps.Invoke();
+            Assert.True(results.Count == 5);
+
+            // Client must not be initialized
+            Assert.NotNull(results[0]);
+            Assert.True((bool)results[0].BaseObject == false);
+
+            // Client must true be initialized
+            Assert.NotNull(results[1]);
+            Assert.True((bool)results[1].BaseObject == true);
+
+            // The grain reference must not be null and of type IManagementGrain
+            Assert.NotNull(results[2]);
+            Assert.True(results[2].BaseObject is IManagementGrain);
+            
+            var statuses = results[3].BaseObject as Dictionary<SiloAddress, SiloStatus>;
+            Assert.NotNull(statuses);
+            Assert.True(statuses.Count > 0);
+            foreach (var pair in statuses)
+            {
+                output.WriteLine("       ######## Silo {0}, status: {1}", pair.Key, pair.Value);
+                Assert.Equal(
+                    SiloStatus.Active,
+                    pair.Value);
+            }
+
+            // Client must be not initialized
+            Assert.NotNull(results[4]);
+            Assert.True((bool)results[4].BaseObject == GrainClient.IsInitialized);            
+        }
+
+        [Fact, TestCategory("BVT"), TestCategory("Tooling")]
+        public void GetGrainTest()
```

Comment:
Yes, you could try to get a `IGuidGrain`, `IStringGrain`, ect. Just to cover the switch statement in `GetGrain.ProcessRecord`